nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py

```diff
@@ -82,142 +83,102 @@ class MegatronTokenLevelEncoderDecoderModule(MegatronModule):
 
     def __init__(
         self,
-        encoder_arch,
-        decoder_arch,
-        vocab_size,
-        hidden_size,
+        encoder_cfg: DictConfig,
```

Comment:
Way cleaner! Nice!

Reply:
Its still hard to keep track of args everywhere, but we're getting close to making it clean :)